### PR TITLE
Vector3<T>

### PR DESCRIPTION
**Pull request by HurricanKai**
See dotnet/runtime#24168
See tannergooding/runtime#1 for tests
`With` methods are not included (yet) because they were only mentioned in a comment.
`public static Vector3 Cross(Vector3<T> left, Vector3<T> right);` has been slightly altered to return `Vector3<T>` because I figured that was simply a typo.

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Numerics/Vector3`1.cs [hunk 1]
```diff
@@ -0,0 +1,693 @@
+﻿using System;
+using System.Globalization;
+using System.Runtime.CompilerServices;
+using System.Security.Cryptography;
+
+namespace System.Numerics
+{
+    public class Vector3<T> : IEquatable<Vector3<T>>, IFormattable
```
Comment (chrarnoldus) [inline src/libraries/System.Private.CoreLib/src/System/Numerics/Vector3`1.cs:8]: Should be struct right?

File: src/libraries/System.Private.CoreLib/src/System/Numerics/Vector3`1.cs [hunk 2]
```diff
@@ -0,0 +1,960 @@
+﻿using System;
```
Comment (tannergooding) [inline src/libraries/System.Private.CoreLib/src/System/Numerics/Vector3`1.cs:1]: Needs the standard license header that all files have.

Comment (tannergooding) [reply src/libraries/System.Private.CoreLib/src/System/Numerics/Vector3`1.cs:1]:
```suggestion
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
// See the LICENSE file in the project root for more information.

﻿using System;
```

File: src/libraries/System.Private.CoreLib/src/System/Numerics/Vector3`1.cs [hunk 3]
```diff
@@ -0,0 +1,960 @@
...
+
+namespace System.Numerics
+{
+    public struct Vector3<T> : IEquatable<Vector3<T>>, IFormattable
+        where T : struct
+    {
+        /// <summary>
+        /// The X component of the vector.
+        /// </summary>
+        public T X { get; }
```
Comment (tannergooding) [inline src/libraries/System.Private.CoreLib/src/System/Numerics/Vector3`1.cs:14]: Missing newline:
```suggestion
        public T X { get; }

```

File: src/libraries/System.Private.CoreLib/src/System/Numerics/Vector3`1.cs [hunk 4]
```diff
@@ -0,0 +1,960 @@
...
+        /// <summary>
+        /// The Y component of the vector.
+        /// </summary>
+        public T Y { get; }
```
Comment (tannergooding) [inline src/libraries/System.Private.CoreLib/src/System/Numerics/Vector3`1.cs:18]: Missing newline:
```suggestion
        public T Y { get; }

```

**Comment by tannergooding** [comment]
Overall LGTM and will unblock the other types being worked on. Thanks!

**Pull request merged**